### PR TITLE
add mycli python deps

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates

#### For security-related PRs

- [ ] The security fix is recorded in `advisories` and `secfixes`

#### For version bump PRs

- [ ] The `epoch` field is reset to 0
- [ ] Patch source: _patch source here_

### CODE REVIEW

File: py3-cffi.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - build-base
+      - python-3.11
+      - python-3.11-dev
+      - py3.11-setuptools
+      - libffi-dev
+
+pipeline:
+  - uses: fetch
+    with:
+      README: 'CONFIRM WITH: curl -L https://files.pythonhosted.org/packages/2b/a8/050ab4f0c3d4c1b8aaa805f70e26e84d0e27004907c5b8ecc1d31815f92a/cffi-1.15.1.tar.gz | sha256sum'
```
Comment (rawlingsj) [inline py3-cffi.yaml:29]: `README` isn't a valid option for the [fetch](https://github.com/chainguard-dev/melange/blob/main/pkg/build/pipelines/fetch.yaml) pipeline.  I'm guessing this comes from the `melange convert` tool when creating python related packages?  We don't validate the schema which would help automatically help catch things like this TBH.

I think it's worth dropping the `README:` option for all these packages as they will become out of date after the next wolfi auto update.

**Review by rawlingsj — CHANGES_REQUESTED** [review]
One more comment - sorry I should have spotted this when adding the last review.  It looks like the `melange convert python` is creating the fetch URLs to pull from https://pypi.org and include a sha in the URL.  This means we can't auto update these package with wolfictl.  

After checking a few of these packages I think all the sources come from github repos?  If we use `git-checkout` repo rather than fetch from https://pypi.org  we can keep these packages up-to-date with our automation.

File: py3-cffi.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - python-3.11
+      - python-3.11-dev
+      - py3.11-setuptools
+      - libffi-dev
+
+pipeline:
+  - uses: fetch
+    with:
+      expected-sha256: d400bfb9a37b1351253cb402671cea7e89bdecc294e8016a707f6d1d8ac934f9
+      uri: https://files.pythonhosted.org/packages/2b/a8/050ab4f0c3d4c1b8aaa805f70e26e84d0e27004907c5b8ecc1d31815f92a/cffi-${{package.version}}.tar.gz
```
Comment (rawlingsj) [inline py3-cffi.yaml:30]: I think we can switch this to remove the sha and auto upgrades should work
```suggestion
      uri: https://foss.heptapod.net/pypy/cffi/-/archive/v${{package.version}}/cffi-v${{package.version}}.tar.gz
```

File: py3-cffi.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+environment:
+  contents:
+    packages:
+      - ca-certificates-bundle
+      - wolfi-base
+      - busybox
+      - build-base
+      - python-3.11
+      - python-3.11-dev
```
Comment (rawlingsj) [inline py3-cffi.yaml:22]: do these need to be python 3.11?  Or can they track the latest python? i.e. dropping the `.11` similar to other examples
```
https://github.com/wolfi-dev/os/blob/d28eeb644d789dd48dfe39aa70fc29ba44672ddb/py3-pluggy.yaml#L19-L21
```

File: py3-pyperclip.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  - name: Python Build
+    runs: python setup.py build
+
+  - name: Python Install
+    runs: python setup.py install --prefix=/usr --root="${{targets.destdir}}"
+
+  - uses: strip
+
+update:
+  enabled: true
```
Comment (rawlingsj) [inline py3-pyperclip.yaml:38]:
```suggestion
  enabled: true
  manual: true # fetch URL contains a sha so will need to be manually updated
```

File: py3-cffi.yaml (hunk 2 above)
Comment (developer-guy) [reply py3-cffi.yaml:30]: thank you!

File: py3-cffi.yaml (hunk 3 above)
Comment (developer-guy) [reply py3-cffi.yaml:22]: let's give it a try, actually this content generated by the `melange convert python` then I didn't change any of the packages afterwards.

**Review by rawlingsj — APPROVED** [review]

**Pull request merged**